src/Uno.UWP/Security/Authentication/Web/WebAuthenticationBrokerProvider.wasm.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Windows.Security.Authentication.Web;
+using Uno.Extensions;
+using Uno.Foundation;
+
+namespace Uno.AuthenticationBroker
+{
+	partial class WebAuthenticationBrokerProvider
+	{
+		private string[] GetCustomSchemes()
+		{
+			var js = $@"Windows.Security.Authentication.Web.WebAuthenticationBroker.getReturnUrl();";
+			var origin = WebAssemblyRuntime.InvokeJS(js);
+
+			return new[] {origin};
+		}
+
+		public async Task<WebAuthenticationResult> AuthenticateAsync(
+			WebAuthenticationOptions options,
+			Uri requestUri,
+			Uri callbackUri,
+			CancellationToken ct)
+		{
+			// options are ignored for now
+
+			// TODO: support ct
+
+			var urlNavigate = WebAssemblyRuntime.EscapeJs(requestUri.OriginalString);
+			var urlRedirect = WebAssemblyRuntime.EscapeJs(callbackUri.OriginalString);
+			string js;
+
+			var timeout = ((long)Timeout.TotalMilliseconds).ToString();
+
+			var useIframe =
+				options.HasFlag(WebAuthenticationOptions.SilentMode) ||
+				!string.IsNullOrEmpty(WinRTFeatureConfiguration.WebAuthenticationBroker.IFrameHtmlId);
+
+			if (useIframe)
+			{
+				var iframeId = WinRTFeatureConfiguration.WebAuthenticationBroker.IFrameHtmlId;
+				js = $@"Windows.Security.Authentication.Web.WebAuthenticationBroker.authenticateUsingIframe(""{iframeId}"", ""{urlNavigate}"", ""{urlRedirect}"", {timeout});";
+			}
+			else
+			{
+				var title = WebAssemblyRuntime.EscapeJs(WinRTFeatureConfiguration.WebAuthenticationBroker.WindowTitle ?? "Sign In");
+
+				var windowWidth = WinRTFeatureConfiguration.WebAuthenticationBroker.WindowWidth;
+				var windowHeight = WinRTFeatureConfiguration.WebAuthenticationBroker.WindowHeight;
+				js = $@"Windows.Security.Authentication.Web.WebAuthenticationBroker.authenticateUsingWindow(""{urlNavigate}"", ""{urlRedirect}"", ""{title}"", {windowWidth}, {windowHeight}, {timeout});";
+			}
+
+			try
+			{
+				var results = (await WebAssemblyRuntime.InvokeAsync(js)).Split(new[] { '|' }, 2);
```

Comment:
use `ct` ?

Reply:
There is not CT in WebAssemblyRuntime.InvokeAsync because there's no such mechanism in Javascript's Promises.